mergify_engine/exceptions.py

```diff
@@ -30,34 +32,31 @@ def __init__(self, ctxt):
         self.ctxt = ctxt
 
 
-RATE_LIMIT_RETRY_MIN = 3
-BASE_RETRY_TIMEOUT = 60
-NOT_ACCESSIBLE_REPOSITORY_MESSAGES = [
-    "Repository access blocked",  # Blocked Github Account or Repo
-    "Resource not accessible by integration",  # missing permission
-]
-
-UNRECOVERABLE_SERVER_ERROR = ["Sorry, this diff is taking too long to generate."]
-
-MISSING_REPOSITORY_DATA_MESSAGE = "Sorry, there was a problem generating this diff. The repository may be missing relevant data."
+RATE_LIMIT_RETRY_MIN: int = 3
+BASE_RETRY_TIMEOUT: int = 60
 
+IGNORED_HTTP_ERRORS: typing.Dict[int, typing.List[str]] = {
+    403: [
+        "Repository access blocked",  # Blocked Github Account or Repo
+        "Resource not accessible by integration",  # missing permission
+    ],
+    422: [
+        "Sorry, there was a problem generating this diff. The repository may be missing relevant data."
+    ],
+    500: ["Sorry, this diff is taking too long to generate."],
+}
 
-def should_be_ignored(exception):
-    if isinstance(exception, http.HTTPClientSideError):
-        if (
-            exception.status_code == 403
-            and exception.message in NOT_ACCESSIBLE_REPOSITORY_MESSAGES
-        ):
-            return True
 
-        elif (
-            exception.status_code == 422
-            and exception.message == MISSING_REPOSITORY_DATA_MESSAGE
-        ):
-            return True
+def should_be_ignored(exception: Exception) -> bool:
+    if isinstance(exception, (http.HTTPClientSideError, http.HTTPServerSideError)):
+        for code, errors in IGNORED_HTTP_ERRORS.items():
+            if exception.status_code == code:
+                for error in errors:
+                    if error in exception.message:
+                        return True
```

Comment:
```suggestion
                for error in IGNORED_HTTP_ERRORS.get(exception.status_code, []):
                    if error in exception.message:
                        return True
```
no?

Reply:
It's same and keeping `if exception.status_code == code:` looks clearer to me.

Reply:
I prefer your finally :p, I just see it's also remove a loop